src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -3470,8 +3470,14 @@ trait Types
       case TypeBounds(_: this.type, _: this.type) => TypeBounds(this, this)
       case oftp                                   => oftp
     }
-    override def lowerBound: Type = bounds.lo
-    override def upperBound: Type = bounds.hi
+    override def lowerBound: Type = underlying.lowerBound match {
+      case _: this.type => this
+      case oftp         => oftp
+    }
```

Comment:
Could very well be a `if (underlying.lowerBound eq this)`, to be sure. Since I just copied what was in the `bounds` method, I chose not to change it from the style choice. But, sure enough, I rather an `if`.

Reply:
Can't it just be `underlying.lowerBound`?